akka-http-tests/src/test/java/akka/http/javadsl/server/JavaRouteTest.java

```diff
@@ -0,0 +1,351 @@
+        runRoute(route, HttpRequest.GET("/uuid").addHeader(Accept.create(MediaRanges.create(MediaTypes.TEXT_XML))))
+            .assertEntity("<id>80a05eee-652e-4458-9bee-19b69dbe1dee</id>")
+            .assertContentType(MediaTypes.TEXT_XML.toContentType(HttpCharsets.UTF_8));
+    }
+    
+    @Test
+    public void request_is_rejected_if_no_marshaller_fits_accepted_type() {
+        runRoute(route, HttpRequest.GET("/uuid").addHeader(Accept.create(MediaRanges.create(MediaTypes.TEXT_PLAIN))))
+            .assertStatusCode(StatusCodes.NOT_ACCEPTABLE);
+    }
+
+    @Test
+    public void first_marshaller_is_picked_and_status_code_applied_if_no_accept_header_present() {
+        runRoute(route, HttpRequest.GET("/uuid"))
+            .assertContentType(MediaTypes.APPLICATION_JSON.toContentType())
+            .assertStatusCode(StatusCodes.FOUND);            
+    }
+    
+    @Test
+    public void exception_handlers_are_applied_even_if_the_route_throws_in_future() {
+        runRoute(route, HttpRequest.GET("/shouldnotfail"))
+            .assertEntity("no problem!");            
+    }
+    
+    @Test
+    public void route_with_required_header_fails_when_header_is_absent() {
+        runRouteUnSealed(route, HttpRequest.GET("/requiredheader"))
+            .assertRejections(Rejections.missingHeader("UUID"));            
+    }
+    
+    @Test
+    public void route_with_required_header_fails_when_header_is_malformed() {
+        runRouteUnSealed(route, HttpRequest.GET("/requiredheader").addHeader(RawHeader.create("UUID", "monkeys")))
+            .assertRejections(Rejections.malformedHeader("UUID", "must be a valid UUID"));            
+    }
+    
+    @Test
+    public void route_with_required_header_succeeds_when_header_is_present_as_RawHeader() {
+        runRoute(route, HttpRequest.GET("/requiredheader").addHeader(RawHeader.create("UUID", "98610fcb-7b19-4639-8dfa-08db8ac19320")))
+            .assertEntity("has header: 98610fcb-7b19-4639-8dfa-08db8ac19320");
+    }
+    
+    @Test
+    public void route_with_required_header_succeeds_when_header_is_present_as_custom_type() {
+        runRoute(route, HttpRequest.GET("/requiredheader").addHeader(new UUIDHeader(UUID.fromString("98610fcb-7b19-4639-8dfa-08db8ac19320"))))
+            .assertEntity("has header: 98610fcb-7b19-4639-8dfa-08db8ac19320");
+    }
+    
+    private final ExceptionHandler xHandler = ExceptionHandler.of(new PFBuilder<Throwable, Route>()
+        .match(IllegalArgumentException.class, x -> complete("no problem!"))
+        .build());
+    
+    private static Function<RequestContext,String> CUSTOM_EXTRACT = ctx -> ctx.getRequest().getHeader("foo").get().value();
+    
+    private CompletionStage<Integer> throwExceptionInFuture() {
+    	return CompletableFuture.supplyAsync(() -> { throw new IllegalArgumentException("always failing"); });
+    }
+    
+    
+    public Route getRoute() {
+        return route(
+            path(segment("hello").slash("world"), () ->
```

Comment:
isn't that already possible? `segment("hello").slash(segment("world"))`.  The `String` variants are just for convenience.